pkg/commands/workdir.go

```diff
@@ -58,6 +58,9 @@ func (w *WorkdirCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile
 		logrus.Infof("Creating directory %s", config.WorkingDir)
 		w.snapshotFiles = append(w.snapshotFiles, config.WorkingDir)
 		return mkdir(config.WorkingDir, 0755)
+	} else {
+		// Force creation of empty cache image to avoid unpacking this layer on reruns
+		w.snapshotFiles = nil
```

Comment:
There's a test failing expecting to get `[]string{}`, not a nil, presumably as a result of this change.

Reply:
:+1: Working on it, I'm seeing some cache misses on the image that I'm trying to build. I'll add some unit and integration tests for caching and FS unpacking.

Reply:
Fixed